### ISSUE TITLE
Make Node a subclass of Tensor?

### ISSUE THREAD

**Issue by repst** [google/TensorNetwork#858]
Having not followed the recent PRs, our group became aware of the new `Tensor` class with the release of 0.4.2. We found that `tensornetwork.ncon`, `tensornetwork.conj` and `tensornetwork.node` take `Tensor` objects and throw errors when passed `Node` objects.  If `Node` inherited from `Tensor` they could share such functionality. I think of a node as a tensor with extra ability to connect to other nodes. 

What do you think of this solution rather than wrapping `Node` around `Tensor` as discussed in #848?

**Comment by mganahl** [comment]
Hi Ryan, great suggestion! I think that would make sense. @Thenerdstation  @alewis, what are your thoughts?

**Comment by alewis** [comment]
It seems fine to me. I don't know how Node works well enough to decide how
hard it would be.

**Comment by mganahl** [comment]
Hey @repst to give you some perspective: we added `Tensor` specifically for use in traditional tensor network calculations where the `Node` API  becomes cumbersome to use. We are currently working to move the DMRG implementation towards a consistent use of `Tensor`.

**Comment by alewis** [comment]
Nevertheless, it would be a nice simplification to the interface if all the
tn. functions worked on Node

**Comment by Thenerdstation** [comment]
Yeah, as long as all of the nodes created by these methods always return with dangling legs, i think that would be fine.
